pkg/broker/broker.go

```diff
@@ -764,14 +774,32 @@ func (a AnsibleBroker) validateDeprovision(instance *apb.ServiceInstance) error
 	return nil
 }
 
-func (a AnsibleBroker) isDeprovisionInProgress(instance *apb.ServiceInstance) (bool, error) {
+func (a AnsibleBroker) isProvisionInProgress(instance *apb.ServiceInstance) (bool, string, error) {
 	allJobs, err := a.dao.GetSvcInstJobsByState(instance.ID.String(), apb.StateInProgress)
 	if err != nil {
-		return false, err
+		return false, "", err
 	}
 
+	var token string
+	proJobs := dao.MapJobStatesWithMethod(allJobs, apb.JobMethodProvision)
+	if len(proJobs) > 0 {
+		token = proJobs[0].Token
+	}
+	return len(proJobs) > 0, token, nil
+}
+
+func (a AnsibleBroker) isDeprovisionInProgress(instance *apb.ServiceInstance) (bool, string, error) {
```

Comment:
This may be more of a nitpick but I am not a big fan of 2 functions that are so identical to each other. I believe there are 2 ways that you could improve this: